spec/controllers/payments_controller_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe "/payments/:id/authorize", type: :request do
```

Comment:
Give that it turned out to be a request spec, I'd say it would be best for it to live under `./spec/requests/` instead of  `./spec/controllers/`. What do you think @kellyzhangg ?

I don't feel qualified to review code quality on this one, so no opinion there.